crates/ide/src/runnables.rs

```diff
@@ -215,8 +253,8 @@ const RUSTDOC_FENCE: &str = "```";
 const RUSTDOC_CODE_BLOCK_ATTRIBUTES_RUNNABLE: &[&str] =
     &["", "rust", "should_panic", "edition2015", "edition2018"];
 
-fn has_runnable_doc_test(fn_def: &ast::Fn) -> bool {
-    fn_def.doc_comment_text().map_or(false, |comments_text| {
+fn has_runnable_doc_test<T: DocCommentsOwner>(def: &T) -> bool {
```

Comment:
I guess this isn't too performance-critical, should we use a trait object here (`def: &dyn DocCommentsOwner`)? https://github.com/rust-analyzer/rust-analyzer/blob/master/docs/dev/style.md#avoid-monomorphization

Reply:
Updated